### DISCUSSION TITLE
Application doesn't respond (discord.py v2.0.0)

### DISCUSSION THREAD

**Discussion by justary27** [Rapptz/discord.py#8360] [Q&A]
I'm migrating to v2.0.0. While trying to send an embed via `@hybrid_command`, I get `Application didn't respond`; the embed gets sent tho!

![image](https://user-images.githubusercontent.com/76696648/185791878-482dafe4-1150-4c30-93a6-59c0dacc7d30.png)

Here's an example command (part of a cog):
```py
@hybrid_command(name="intro")
async def send_name(self, ctx):
  embed = Embed(title="Intro", description="Hello from bot!")
  await ctx.send(embed=embed)
```
I have tried `ctx.respond, ctx.reply` (from StackOverflow).

**Comment by jackthenewbie** [comment]
did ya try to add ``ctx.defer()``?

**Reply by justary27** [reply]
Well I tried this before coming back here today and yes, `ctx.defer()` does the job. Not sure if it's good practice as Rapptz has added a thumbs down :) Anyways, thanks!

**Reply by Rapptz** [reply]
It's a thumbs down because `ctx.send` responds to the interaction already. If this actually solved the issue then, on top of the syntax error in the original snippet, it wasn't the actual code being used.

**Reply by justary27** [reply]
Well the original code is this:
```py
@hybrid_command(name="invite", with_app_command=True, description="Receive the bot's invite link in your DM.")
    async def invite_bot(self, ctx):
        """
        The invite command, instantly generates bot's invite link.
        """

        invite_url = discord.utils.oauth_url(client_id="CLIENT_ID")
        embed = discord.Embed(title="Invite link", description="", color=discord.Color.from_rgb(255, 175, 175))
        embed.add_field(name="Here's my invite link!",
                        value=f"[**Click here to invite**]({invite_url})")
        await ctx.send(embed=embed)
```
I added `ctx.defer()`, after `ctx.send()` and it solves the problem, dunno why tho.

**Reply by jackthenewbie** [reply]
probably something else in his code actually caused the problem, dont know where, but hey it works:)

**Comment by staciax** [comment]
![image](https://user-images.githubusercontent.com/69411527/185802101-9e57408d-f76e-418b-8a4c-4e92957abfda.png)
what wait? `=`

**Reply by justary27** [reply]
Well, it was a typo while typing here :)

**Comment by staciax** [comment]
hybird command I'm not sure if it needs to sync or not.
```py

# in class bot
 async def setup_hook(self):
        await self.tree.sync()
```

**Reply by lol219** [reply]
Don't put syncing in startup, it's just make everything worse

**Reply by staciax** [reply]
setup_hook is special method

**Reply by SawshaDev** [reply]
Auto syncing should not be done. just because it's in setup_hook does not make it good to auto sync.
you should only sync in a command just like a regular text based commands

**Reply by justary27** [reply]
Well, you don't need to do that, the problem isn't that the command isn't registered!

**Comment by lol219** [comment]
Did u fixed ur problem @just-ary27
